### PR TITLE
validate IP address while configuring admiral

### PR DESCRIPTION
**Pull request by deepikasl**
https://github.com/Shippable/admiral/issues/1490

### CODE REVIEW

File: common/scripts/lib/_helpers.sh [hunk 1]
```diff
@@ -312,8 +312,15 @@ __set_admiral_ip() {
   __process_success "Please enter your current IP address. This will be the address at which you access the installer webpage. Type D to set default (127.0.0.1) value."
   read response
 
+  local rx='([1-9]?[0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])'
```
Comment (ric03uec) [inline common/scripts/lib/_helpers.sh]: lets move this into a separate function called `_validate_ip`. use that function anywhere where validation is required

Comment (deepikasl) [reply common/scripts/lib/_helpers.sh]: ok

File: common/scripts/lib/_helpers.sh [hunk 2]
```diff
@@ -305,6 +305,18 @@ __set_secret_key() {
...
 
+_validate_ip() {
+  local ip=$1
+  local step=$2
+  local rx='([1-9]?[0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])'
+  if [[ $ip =~ ^$rx\.$rx\.$rx\.$rx$ ]]; then
+    return
+  else
+    __process_error "Invalid response, please enter valid IP address"
+    $step
```
Comment (ric03uec) [inline common/scripts/lib/_helpers.sh]: why is this required?
- this should be a simple function that returns true/false and the caller needs to decide what to with the return value. this implementation is too complex for a simple problem
